src/plugins/intel_gpu/include/intel_gpu/runtime/memory.hpp

```diff
@@ -180,14 +180,28 @@ inline std::vector<T> read_vector(cldnn::memory::ptr mem, const cldnn::stream& s
         switch (mem->get_layout().data_type) {
             case data_types::i32: {
                 auto p_mem = reinterpret_cast<int32_t*>(mem->buffer_ptr());
-                for (size_t i = 0; i < mem->count(); i++) {
+                for (size_t i = 0; i < mem->count(); ++i) {
                     out_vecs.push_back(static_cast<T>(p_mem[i]));
                 }
                 break;
             }
             case data_types::i64: {
                 auto p_mem = reinterpret_cast<int64_t*>(mem->buffer_ptr());
-                for (size_t i = 0; i < mem->count(); i++) {
+                for (size_t i = 0; i < mem->count(); ++i) {
+                    out_vecs.push_back(static_cast<T>(p_mem[i]));
+                }
+                break;
+            }
+            case data_types::f16: {
+                auto p_mem = reinterpret_cast<uint16_t*>(mem->buffer_ptr());
+                for (size_t i = 0; i < mem->count(); ++i) {
+                    out_vecs.push_back(static_cast<T>(p_mem[i]));
```

Comment:
In that case static_cast is not correct - you need to convert half_to_float and then cast to final type (if it's not fp16). Moreover, maybe we need a check that `T` is floating point when `mem->get_layout().data_type` is floating point type to avoid some unexpected things.

Reply:
Fixed